compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/types/Types.java

```diff
@@ -0,0 +1,201 @@
+     * The module is determined by the provided org, module name and the version.
+     *
+     * @param org           The organization of the looking up module
+     * @param moduleName    The name of the looking up module
+     * @param version       The version of the looking up module
+     * @return A {@link Map} of the user defined type symbols
+     */
+    public Optional<Map<String, Symbol>> typesInModule(String org, String moduleName, String version) {
+        PackageID packageID = new PackageID(Names.fromString(org), Names.fromString(moduleName),
+                Names.fromString(version));
+
+        return getTypeDefSymbolsInModule(packageID);
+    }
+
+    public static Types getInstance(CompilerContext context) {
+        Types types = context.get(TYPES_KEY);
+        if (types == null) {
+            types = new Types(context);
+        }
+
+        return types;
+    }
+
+    private Optional<Symbol> getTypeDefByName(PackageID packageID, String typeDefName) {
+        BPackageSymbol packageSymbol = packageCache.getSymbol(packageID);
+        if (packageSymbol == null) {
+            return Optional.empty();
+        }
+
+        SymbolEnv pkgEnv = symbolTable.pkgEnvMap.get(packageSymbol);
+        Scope.ScopeEntry entry = pkgEnv.scope.lookup(Names.fromString(typeDefName));
+        if (entry != null) {
+            if (isValidTypeDef(entry.symbol)) {
+                return Optional.ofNullable(symbolFactory.getBCompiledSymbol(entry.symbol, typeDefName));
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<Map<String, Symbol>> getTypeDefSymbolsInModule(PackageID packageID) {
+        BPackageSymbol packageSymbol = packageCache.getSymbol(packageID);
+        if (packageSymbol == null) {
+            return Optional.empty();
+        }
+
+        SymbolEnv pkgEnv = symbolTable.pkgEnvMap.get(packageSymbol);
+        Scope pkgEnvScope = pkgEnv.scope;
+        if (pkgEnvScope != null && pkgEnvScope.entries != null) {
+            Map<String, Symbol> typeDefSymbols = new HashMap<>();
+            for (Scope.ScopeEntry scopeEntry : pkgEnvScope.entries.values()) {
+                BSymbol bSymbol = scopeEntry.symbol;
+                if (isValidTypeDef(bSymbol)) {
+                    String typeDefName = bSymbol.getOriginalName().getValue();
+                    typeDefSymbols.put(typeDefName, symbolFactory.getBCompiledSymbol(bSymbol, typeDefName));
+                }
+            }
+
+            return Optional.of(Collections.unmodifiableMap(typeDefSymbols));
+        }
+
+        return Optional.empty();
+    }
+
+    private boolean isValidTypeDef(BSymbol bSymbol) {
+        return bSymbol != null && bSymbol.getOrigin() != SymbolOrigin.VIRTUAL
+                && (bSymbol.getKind() == SymbolKind.TYPE_DEF
+                || bSymbol.getKind() == SymbolKind.CONSTANT
+                || bSymbol.getKind() == SymbolKind.ENUM
+                || Symbols.isFlagOn(bSymbol.flags, Flags.CLASS));
```

Comment:
What's the need for checking against symbol kind for the rest and using the flags for classes? Also, would this disregard classes created for services?

Reply:
@pubudu91 
The `SymbolKind.OBJECT` is being used as the symbol kind of `BallerinaClassSymbol`s. 
https://github.com/ballerina-platform/ballerina-lang/blob/fcf27a28819858421f3a1a7820d81b31e1f5e3e8/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaClassSymbol.java#L131-L133

And yes, the classes created for services do have a `VIRTUAL` symbol origin. Therefore, they're disregarded. 
Will add a test case to assure this as well.